### PR TITLE
Rebased support for Aligned Function Attribute

### PR DESCRIPTION
**Pull request by shubhamnarlawar77**

### CODE REVIEW

**Comment by shubhamnarlawar77** [comment]
'@marxin: review request'

**Comment by shubhamnarlawar77** [comment]
Aligned attribute is output in `OutputAttributes` as it has some computation.

File: src/Function.cpp [hunk 1]
```diff
@@ -119,6 +119,25 @@ MultiValuedFunctionAttribute::OutputAttributes(std::ostream &out)
...
+{
+}
+
+void
+AlignedFunctionAttribute::OutputAttributes(std::ostream &out)
+{
+	if(rnd_flipcoin(attribute_probability)){
+	int power = rnd_upto(16);
+		if(!attr_emitted){
+			out << " __attribute__((" << "aligned(" << (1 << power) << ")";
```
Comment (marxin) [inline src/Function.cpp]: This should be handled only at one place, which is probably `FunctionAttribute::OutputAttribute`.

File: src/Function.cpp [hunk 2]
```diff
@@ -119,6 +119,25 @@ MultiValuedFunctionAttribute::OutputAttributes(std::ostream &out)
...
+	: FunctionAttribute(name, prob)
+{
+}
+
+void
+AlignedFunctionAttribute::OutputAttributes(std::ostream &out)
+{
+	if(rnd_flipcoin(attribute_probability)){
+	int power = rnd_upto(16);
+		if(!attr_emitted){
```
Comment (marxin) [inline src/Function.cpp]: Bad indentation here.

**Review by marxin — CHANGES_REQUESTED** [review]

File: src/Function.cpp [hunk 3]
```diff
@@ -95,6 +95,17 @@ FunctionAttribute::OutputAttribute(std::ostream &out, string option)
 		out << ", " << attribute << option;
 }
 
+void
+FunctionAttribute::OutputAttribute(std::ostream &out, int option)
```
Comment (marxin) [inline src/Function.cpp]: Do not create a new override. Please make a string in the caller and call the version that takes `string` as argument.

**Review by marxin — CHANGES_REQUESTED** [review]

File: src/Function.cpp (hunk 3 above)
Comment (shubhamnarlawar77) [reply src/Function.cpp]: Done.

File: src/Function.cpp [hunk 4]
```diff
@@ -119,6 +119,18 @@ MultiValuedFunctionAttribute::OutputAttributes(std::ostream &out)
...
 
+AlignedFunctionAttribute::AlignedFunctionAttribute(string name, int prob)
+	: FunctionAttribute(name, prob)
+{
+}
+
+void
+AlignedFunctionAttribute::OutputAttributes(std::ostream &out)
+{
+	int power = rnd_upto(16);
```
Comment (marxin) [inline src/Function.cpp]: There's missing a flip coin.

**Review by marxin — CHANGES_REQUESTED** [review]

File: src/Function.cpp (hunk 4 above)
Comment (shubhamnarlawar77) [reply src/Function.cpp]: > There's missing a flip coin.

Done.

Comment (shubhamnarlawar77) [reply src/Function.cpp]: Is pull request ready to merge?

**Comment by marxin** [comment]
Now it's fine, thank you!

**Comment by shubhamnarlawar77** [comment]
> Now it's fine, thank you!

Is this a yes to merge pull request?

**Comment by marxin** [comment]
> Is this a yes to merge pull request?

Yes :)

**Pull request merged**